src/components/DatePicker/components/Day/Day.tsx

```diff
@@ -16,84 +13,72 @@ export interface DayProps {
   inHoveringRange?: boolean;
   disabled?: boolean;
   onClick?(day: Date): void;
-  onHover?(day: Date): void;
-  onFocus?(day: Date): void;
+  onHover?(day?: Date): void;
```

Comment:
Day is not a public member of the Polaris API, so the only consumers are internal, and this doesn't throw any type errors so that's not a concern.

I think we're safe to make any changes we want to private parts of our API. Do you disagree?

Reply:
Good point on this not actually being public, so the breaking change thing is moot.

It looks like we can keep `onFocus?(day: Date)` without any extra changes needed so that'd be easy to do.

I was thinking we could remove the onHover from the empty day but it looks like that has knock of effects for how selecting a range (with `<DatePicker allowRange>`) works so it looks like we need to keep that.

Reply:
I'm also wary of making too many changes in this PR, I'm already making a few changes to the flow here and there and I'm not super familiar with the Datepicker so I'm trying to keep everything roughly the same